File: android/src/main/java/org/ligi/passandroid/helper/MoveHelper.java
package org.ligi.passandroid.helper;

import android.app.Activity;
import android.support.design.widget.Snackbar;
import android.view.View;

import org.ligi.passandroid.R;
import org.ligi.passandroid.model.Pass;
import org.ligi.passandroid.model.PassClassifier;

import java.lang.String;

public class MoveHelper {
    public static void moveWithUndoSnackbar(final PassClassifier passClassifier, final Pass pass, String topic, final Activity activity) {
        final String oldTopic = passClassifier.getTopic(pass);

        Snackbar.make(activity.getWindow().getDecorView().findViewById(R.id.fam), String.format(activity.getString(R.string.notification_moved_to), topic), Snackbar.LENGTH_LONG)
                .setAction(activity.getString(R.string.action_undo), new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        passClassifier.moveToTopic(pass, oldTopic);
                    }
                })
                .show();
        passClassifier.moveToTopic(pass, topic);
    }
}

File: android/src/main/java/org/ligi/passandroid/functions/MoveHelper.kt
package org.ligi.passandroid.functions

import android.app.Activity
import android.support.design.widget.Snackbar
import org.ligi.passandroid.R
import org.ligi.passandroid.model.PassClassifier
import org.ligi.passandroid.model.pass.Pass

fun moveWithUndoSnackbar(passClassifier: PassClassifier, pass: Pass, topic: String, activity: Activity) {
    val oldTopic = passClassifier.getTopic(pass, "")

    Snackbar.make(activity.window.decorView.findViewById(R.id.fam), "Pass moved to " + topic, Snackbar.LENGTH_LONG)
            .setAction(R.string.undo) { passClassifier.moveToTopic(pass, oldTopic) }
            .show()
    passClassifier.moveToTopic(pass, topic)
}
